docs/10-WebServices.md

```diff
@@ -67,6 +107,54 @@ class ApiHelper extends \Codeception\Module
 
 The same way you can receive request parameters and headers.
 
+### Validate JSON structures
+
+It is pretty common for API tests to not only validate the received data but to check the structure of the response. Response data is not something constant, and may change on each request, however the JSON/XML structure should kept the same for an API version. In order to check response structure REST module have some useful methods. 
+
+If we expect JSON response to be received we can check its structure with [JSONPath](http://goessner.net/articles/JsonPath/). It looks and sounds like XPath but is designed to work with JSON data, however we can convert JSON into XML and use XPath to validate the structure. Both approaches are valid and can be used in REST module:
+
+```php
+<?php
+$I = new ApiTester($scenario);
+$I->wantTo('validate structure of GitHub api responses');
+$I->sendGET('/users');
+$I->seeResponseIsJson();
+$I->seeResponseJsonMatchesJsonPath('$[0].user.login');
+$I->seeResponseJsonMatchesXpath('//user/login');
+?>
+```
+
+### Testing XML Responses
+
+In case your REST API works with XML format you can use similar methods to test its data and structure. 
+There is `seeXmlResponseIncludes` method to match inclusion of XML parts in response, and `seeXmlResponseMatchesXpath` to validate its structure.
+
+```php
+<?php
+use Codeception\Util\Xml as XmlUtils;
+
+$I = new ApiTester($scenario);
+$I->wantTo('validate structure of GitHub api responses');
+$I->sendGET('/users.xml');
+$I->seeResponseIsXml();
+$I->seeXmlResponseMatchesXpath('//user/login');
+$I->seeXmlResponseIncludes(XmlUtils::toXml(
+		'user' => [
+			'name' => 'davert',
+			'email' => 'davert@codeception.com',
+			'status' => 'inactive'
+	]
+));
+?>
+```
+
+We are using XmlUtils class which allows us to build XML structures in a clean manner. `toXml` method may accept string, array and returns \DOMDocument instance. If your XML contains attributes and so can't be represented as PHP array you can create XML using [XmlBulder](http://codeception.com/docs/reference/XmlBuilder) class. We will take a look at it a bit more in next section. 
```

Comment:
**The** `toXml` method may accept **a** string**~~,~~ or** array

represented as **a** PHP array you can create XML using **the**